brian2genn/device.py

```diff
@@ -1649,6 +1649,38 @@ def process_state_monitors(self, directory, state_monitors, writer):
 
             self.state_monitor_models.append(sm)
 
+    def consolidate_pull_operations(self, run_regularly_operations):
+        def append(adict, name, step):
+            if name in adict:
+                adict[name].append(step)
+            else:
+                adict[name] = [step]
```

Comment:
I'd prefer to use `defaultdict` from `collections`, i.e. use:
```python
models_start = collections.defaultdict(list)
```
and then you can directly do
```python
models_start[key].append(...)
```
without having to worry whether it is the first time you are using that key.